### ISSUE TITLE
Equivalent changes not showing up in report with reportSeverity set to equivalent

### ISSUE THREAD

**Issue by Singleton06** [revapi/revapi#155]
I am getting an issue where the revapi:report goal does not seem to be honoring the reportSeverity property.  I have set it to `equivalent` and it seems that it does not honor that setting.

This project shows the issue:
https://github.com/Singleton06/revapi-issue-reproduction

Specific things to note...
Here is the report that gets generated:
![image](https://user-images.githubusercontent.com/2235110/57309226-4f15e980-70ad-11e9-80a8-7a86de5cbc30.png)

Report configuration:
https://github.com/Singleton06/revapi-issue-reproduction/blob/master/pom.xml#L94

Differences between the projects:
v2 has a completely new class: https://github.com/Singleton06/revapi-issue-reproduction/blob/master/v2/src/main/java/com/example/NewClass.java

v2 has made the "doNothing" method public: [v1](https://github.com/Singleton06/revapi-issue-reproduction/blob/master/v1/src/main/java/com/example/TestingClass.java) [v2](https://github.com/Singleton06/revapi-issue-reproduction/blob/master/v2/src/main/java/com/example/TestingClass.java)

v2 has made a new method on the `TestingInterface`:
https://github.com/Singleton06/revapi-issue-reproduction/blob/master/v2/src/main/java/com/example/TestingInterface.java#L7

The only thing that is showing though, is the last item.

**Comment by metlos** [comment]
Thanks for the report! Unfortunately I will only be able to take a look at it (or assist you in fixing it ;) ) some time next week.

**Comment by metlos** [comment]
The revapi configuration in your sample project is specific to a concrete execution of the `check` goal and hence doesn't apply to the `report` goal. If you move the configuration to the plugin definition level then it will be shared for all goals and `report` will start behaving as expected.

**Comment by Singleton06** [comment]
The line that I linked has the reportSeverity within the reporting section.  Am I missing something?  I would assume that the configuration specified in the reporting section would be what is used for the reporting.
https://github.com/Singleton06/revapi-issue-reproduction/blob/master/pom.xml#L94

**Comment by metlos** [comment]
I assume you were executing the plugin directly, e.g. `mvn revapi:report` at which point it is using the config from the `build` section. I tried running `mvn site` on your test project and it failed with:
```
Execution default-site of goal org.apache.maven.plugins:maven-site-plugin:3.3:site failed: A required class was missing while executing org.apache.maven.plugins:maven-site-plugin:3.3:site: org/apache/maven/doxia/siterenderer/DocumentContent
```

**Comment by metlos** [comment]
k, I'm reopening, something indeed seems to be off :)

**Comment by metlos** [comment]
So here's what I'm doing:
I've updated the pom.xml to include:
```xml
<plugin>
  <artifactId>maven-site-plugin</artifactId>
  <version>3.7</version>
</plugin>
```
and then:
1. Clean the previous state:
```
$ rm -Rf ~/.m2/repository/com/example/revapi
$ mvn clean
```
2. Trying to generate a site now gives:
```
$ mvn site
...
[INFO] ---------------------< com.example.revapi:example >---------------------
[INFO] Building example 2.0.0-SNAPSHOT                                    [3/3]
[INFO] --------------------------------[ jar ]---------------------------------
...
[INFO] Generating "API Change Analysis" report --- revapi-maven-plugin:0.11.1:report
[WARNING] Failed to resolve old artifacts: Failed to find a version of artifact 'com.example.revapi:example:jar:RELEASE' that would correspond to an expression '^.*(?<!-SNAPSHOT)$'. The versions found were: []. The API analysis will proceed comparing the new archives against an empty archive.
[WARNING] Failed to resolve new artifacts: Could not find artifact com.example.revapi:example:jar:2.0.0-SNAPSHOT. The API analysis will not proceed.
[WARNING] Could not determine the artifacts to compare. If you're comparing the currently built version, have you run the package goal?
...
```
and the generated site looks like:
![image](https://user-images.githubusercontent.com/19999/57714070-ff6a8b80-7673-11e9-8767-6f2023d157ff.png)
3. K, that's expected, we need to run the package goal first to have the artifacts in `target/`.
```
$ mvn package
...
$ mvn site
...
[INFO] Comparing [] against [com.example.revapi:example:jar:2.0.0-SNAPSHOT] (including their transitive dependencies).
...
```
That's slightly better but Maven failed to find our `1.0.0` version even though it is ready in the `target/`. But because `mvn site` was run standalone, the `target/` is not actually part of the reactor, so Maven doesn't look there during artifact resolution. The report we get now is:
![image](https://user-images.githubusercontent.com/19999/57714396-bc5ce800-7674-11e9-8617-61ecf214ecbd.png)
While slightly confusing, it precisely corresponds to the configuration of the reporting (e.g. reports everything because of the `equivalent` report severity configured for reporting) and the message it gave during the report generation. Maven was not able to resolve the old version and so we're only working with the new archive.
4. Let's try force `1.0.0` into the reactor. We can do it in 2 ways. Either we can run `mvn package site` which will make even `1.0.0` part of the reactor and the report will look like expected:
![image](https://user-images.githubusercontent.com/19999/57714749-75bbbd80-7675-11e9-95d1-0e33a3dd2022.png)
or we can run `mvn install` first and only then `mvn site`. This will result in the exactly same report as above, because Maven will resolve the `1.0.0` from the local repo to which we installed the artifact to in the `mvn install` run.

Nothing in there looks wrong to me in the end. Could you please try to prepare repro steps that get you from the clean slate to the unexpected report?

**Comment by Singleton06** [comment]
You will notice that the "TestingClass" which has increased the visibility of the method is not on your report and should be based on the "equivalent" setting.

**Comment by metlos** [comment]
That would be because equivalent changes are not included in the report at all :man_facepalming:

**Comment by metlos** [comment]
Fixed in `revapi-maven-plugin-0.11.2`
